test/pthread/test_pthread_proxy_to_pthread.c

```diff
@@ -5,14 +5,43 @@
  * found in the LICENSE file.
  */
 
-#include <stdio.h>
-#include <emscripten.h>
 #include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include <emscripten/em_asm.h>
+#include <emscripten/eventloop.h>
+#include <emscripten/threading.h>
+
+_Atomic bool done_callback = false;
+
+void main_thread_callback(void* user_data) {
+  printf("main_thread_callback called\n");
+  done_callback = true;
+}
+
+void async_on_main() {
+  int is_worker = EM_ASM_INT(return ENVIRONMENT_IS_WORKER);
+  printf("run_on_main: ENVIRONMENT_IS_WORKER: %d\n", is_worker);
+  // Verify that we can do async work here on the main thread
+  // without causing the runtime to exit.
+  emscripten_set_immediate(main_thread_callback, NULL);
+}
+
+int main() {
+  int is_worker = EM_ASM_INT(return ENVIRONMENT_IS_WORKER);
+  printf("ENVIRONMENT_IS_WORKER: %d\n", is_worker);
+  assert(is_worker);
+
+  emscripten_sync_run_in_main_runtime_thread(EM_FUNC_SIG_V, async_on_main);
```

Comment:
the function is named `async_on_main` but it's run via `emscripten_sync_run_in_main_runtime_thread`?

Reply:
The job of this function is to kick of some async work.. but the function itself is synchronous.

Reply:
I will rename it